lib/g_searcher_web/validators/param_validator.ex

```diff
@@ -0,0 +1,18 @@
+defmodule GSearcherWeb.Validators.ParamValidator do
+  def validate(params, for: params_module) do
+    params
+    |> params_module.changeset()
+    |> add_validate_action()
+    |> handle_changeset()
+  end
+
+  defp handle_changeset(%Ecto.Changeset{changes: changes, valid?: true}),
+    do: {:ok, changes}
+
+  defp handle_changeset(changeset) do
+    {:error, :invalid_params, changeset}
+  end
+
+  defp add_validate_action(changeset),
+    do: %{changeset | action: :validate}
```

Comment:
Thinking maybe we could use `apply_action` - https://hexdocs.pm/ecto/Ecto.Changeset.html#apply_action/2

Reply:
Yeah, when I checked what the method did, it does the same thing as what we're doing here 😂 
I've updated it to that 👍